rellic/AST/ExprCombine.cpp

```diff
@@ -82,9 +82,9 @@ class AddrOfArraySubscriptRule : public InferenceRule {
 class DerefAddrOfRule : public InferenceRule {
  public:
   DerefAddrOfRule()
-      : InferenceRule(unaryOperator(stmt().bind("deref"), hasOperatorName("*"),
-                                    has(unaryOperator(hasOperatorName("&"))))) {
-  }
+      : InferenceRule(unaryOperator(
+            stmt().bind("deref"), hasOperatorName("*"),
+            has(ignoringParenCasts(unaryOperator(hasOperatorName("&")))))) {}
```

Comment:
Can it happen that `ignoringParenCasts` will accept an outermost `(uintptr_t)`, or does that imply something structurally unacceptable? That is, we're in a deref, therefore we must be dereferencing an address-of.

Reply:
As far as I know, any C-style cast will disqualify the match, since `ignoringParenCasts` does not ignore C-style cast expressions.

> That is, we're in a deref, therefore we must be dereferencing an address-of.

This is correct. An address-of that can be wrapped in cast expressions that `ignoringParenCasts` catches.